web/src/components/auth/utils.js

```diff
@@ -0,0 +1,194 @@
+import axios from "axios";
+import {
+  apiUrlCreateJWTToken,
+  apiUrlRegister,
+  apiUrlLoadUser,
+  apiUrlVerifyToken,
+  apiUrlVerifyActivation,
+} from "../../api.util";
+
+/**
+ * Shape of User stored in database
+ * @typeof User
+ * @property {string} first_name
+ * @property {string} last_name
+ * @property {string} email
+ * @property {int} id
+ */
+
+/**
+ * Shape of JWT Web Token pair
+ * @typeof WebToken
+ * @property {string} refresh
+ * @property {string} access
+ */
+
+/**
+ * Stores access token in Local Storage
+ * @param {string} access the access token
+ */
+const setAccessTokenOnSuccess = (access) => {
+  localStorage.setItem("access", access);
+};
+
+/**
+ * Sets access token to null on failure in Local Storage
+ *
+ */
+const setTokensOnFailure = () => {
+  localStorage.removeItem("access");
+  localStorage.removeItem("refresh");
+};
+
+/**
+ * Logs-in a user
+ * @param {string} email the user's email
+ * @param {string} password the user's password
+ * @returns {boolean}
+ */
+export const login = async (email, password) => {
+  const options = {
+    headers: {
+      "Content-Type": "application/json",
+    },
+  };
+  const body = JSON.stringify({ email, password });
+
+  try {
+    const res = await axios.post(apiUrlCreateJWTToken(), body, options);
+    let isAuthenticated = false;
+    if (res && res.data && res.data.access) {
+      isAuthenticated = checkAuthenticated(res.data.access);
+    }
+    if (isAuthenticated) {
+      setAccessTokenOnSuccess(res.data.access);
+      loadUser(res.data.access);
+    }
+    return isAuthenticated;
+  } catch (err) {
+    console.error("Logging-in was unsuccessful", err);
+    setTokensOnFailure();
+  }
+};
```

Comment:
`loadUser`, `setAccessTokenOnSuccess`, `setAccessTokenOnFailure`, and `checkAuthenticated` each has its own side effect. I recommended moving them from this function and up to the same level/location where `login` is being called.